### PR TITLE
Add my kitchen page

### PR DESCRIPTION
**Pull request by Airboy98**
created frontend/src/components and frontend/src/pages/ folders and transplanted and modified (detailed below) user-inventory prototype code to frontend/src/pages/Kitchen.js file and embedded the page component in frontend/src/App.js 

touched up a few bits of the My Kitchen table:
1. switched CSS color theme (a little closer to the overall red/yellow smoothie color theme we like, still needs workshopping)
2. decoupled the InputNumber components in the Quantity and Servings columns of the DataTable component
3. changed the step of InputNumber component in DataTable component Servings column to step by 1 instead of .25
4. created deleteHandler function that's currently used as a dummy trash icon delete button template in the delete column of DataTable and modified its CSS styling

### CODE REVIEW

File: frontend/src/pages/Kitchen.js [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+    return (
+      <InputNumber
+        value={rowData.servings}
+        onValueChange={(e) => onServingsChange(rowData, e)}
+        showButtons
+        buttonLayout="horizontal"
+        step={1}
+        decrementButtonClassName="p-button-danger"
+        incrementButtonClassName="p-button-success"
+        inputClassName="quantity-input" // Apply custom class for input field
```
Comment (sethbonnie) [inline frontend/src/pages/Kitchen.js]: These comments are just restating what the code says, can you remove?

File: frontend/src/pages/Kitchen.js [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+      />
+    );
+  };
+
+  //--------------FUNCTION FOR TABLE SERVINGS TEMPLATE--------------
```
Comment (sethbonnie) [inline frontend/src/pages/Kitchen.js]: What's the purpose of these headers? Can they be removed?

File: frontend/src/pages/Kitchen.js [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+      name: "Sugar",
+      category: "sweetener",
+      size: "2lb bag",
+      quantity: 3,
+      servings: 1,
+    },
+    // Add more products as needed
+  ]);
+
+  //--------------STATIC ARRAY OF HARDCODED DATA FOR TABLE (eventually transplant to .json file)--------------
```
Comment (sethbonnie) [inline frontend/src/pages/Kitchen.js]: Probably can remove this

File: frontend/src/pages/Kitchen.js [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+              <AutoComplete
+                //--------------TEXT FIELD FOR ITEM NAME WITH AUTOCOMPLETE--------------
+                value={selectedItem}
+                onChange={(e) => setSelectedItem(e.value)}
+                suggestions={filteredItems}
+                completeMethod={search}
+                field="label"
+                optionGroupLabel="label"
+                optionGroupChildren="items"
+                // optionGroupTemplate={groupedItemTemplate}
```
Comment (sethbonnie) [inline frontend/src/pages/Kitchen.js]: Can you remove unused code like this? It adds visual clutter while doing nothing.

File: frontend/src/pages/Kitchen.js [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+                buttonLayout="horizontal"
+                step={0.25}
+                decrementButtonClassName="p-button-danger"
+                incrementButtonClassName="p-button-success"
+
+                // incrementButtonIcon="+"
+                // decrementButtonIcon="-"
+                // mode="currency"
+                // currency="USA"
+                //--------------INPUT FIELD FOR ITEM QUANTITY--------------
```
Comment (sethbonnie) [inline frontend/src/pages/Kitchen.js]: Same here, these unused lines should be removed.

**Review by sethbonnie — CHANGES_REQUESTED** [review]

**Review by sethbonnie — DISMISSED** [review]

**Review by nikitakabir — APPROVED** [review]

**Pull request merged**